chirp/drivers/baofeng_uv17Pro.py

```diff
@@ -0,0 +1,1427 @@
+            mem.name += str(char)
+        mem.name = mem.name.rstrip()
+
+        if not _mem.scan:
+            mem.skip = "S"
+
+        levels = self.POWER_LEVELS
+        try:
+            mem.power = levels[_mem.lowpower]
+        except IndexError:
+            mem.power = levels[0]
+
+        mem.mode = _mem.wide and "NFM" or "FM"
+
+        dtcs_pol = ["N", "N"]
+
+        if _mem.txtone in [0, 0xFFFF]:
+            txmode = ""
+        elif _mem.txtone >= 0x0258:
+            txmode = "Tone"
+            mem.rtone = int(_mem.txtone) / 10.0
+        elif _mem.txtone <= 0x0258:
+            txmode = "DTCS"
+            if _mem.txtone > 0x69:
+                index = _mem.txtone - 0x6A
+                dtcs_pol[0] = "R"
+            else:
+                index = _mem.txtone - 1
+            mem.dtcs = self.DTCS_CODES[index]
+        else:
+            LOG.warn("Bug: txtone is %04x" % _mem.txtone)
+
+        if _mem.rxtone in [0, 0xFFFF]:
+            rxmode = ""
+        elif _mem.rxtone >= 0x0258:
+            rxmode = "Tone"
+            mem.ctone = int(_mem.rxtone) / 10.0
+        elif _mem.rxtone <= 0x0258:
+            rxmode = "DTCS"
+            if _mem.rxtone >= 0x6A:
+                index = _mem.rxtone - 0x6A
+                dtcs_pol[1] = "R"
+            else:
+                index = _mem.rxtone - 1
+            mem.rx_dtcs = self.DTCS_CODES[index]
+        else:
+            LOG.warn("Bug: rxtone is %04x" % _mem.rxtone)
+
+        if txmode == "Tone" and not rxmode:
+            mem.tmode = "Tone"
+        elif txmode == rxmode and txmode == "Tone" and mem.rtone == mem.ctone:
+            mem.tmode = "TSQL"
+        elif txmode == rxmode and txmode == "DTCS" and mem.dtcs == mem.rx_dtcs:
+            mem.tmode = "DTCS"
+        elif rxmode or txmode:
+            mem.tmode = "Cross"
+            mem.cross_mode = "%s->%s" % (txmode, rxmode)
+
+        mem.dtcs_polarity = "".join(dtcs_pol)
+
+        mem.extra = RadioSettingGroup("Extra", "extra")
+
+        rs = RadioSetting("bcl", "BCL",
+                          RadioSettingValueBoolean(_mem.bcl))
+        mem.extra.append(rs)
+
+        if (_mem.pttid != 0xFF):
+            rs = RadioSetting("pttid", "PTT ID",
+                              RadioSettingValueList(self.PTTID_LIST,
+                                                    self.PTTID_LIST[
+                                                        _mem.pttid]))
+            mem.extra.append(rs)
+
+        if (_mem.scode != 0xFF):
+            rs = RadioSetting("scode", "S-CODE",
+                              RadioSettingValueList(self.SCODE_LIST,
+                                                    self.SCODE_LIST[
+                                                        _mem.scode]))
+            mem.extra.append(rs)
+
+        return mem
+
+    def set_memory(self, mem):
+        _mem = self._memobj.memory[mem.number]
+
+        _mem.set_raw("\x00"*16 + "\xff" * 16)
```

Comment:
These need to be `bytes` now. I haven't enabled strict mode for this yet, but it's coming.

Reply:
Fixed